### PR TITLE
chore: Updating the settings icon throughout the product

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Updating the settings icon throughout the product to use the below icon:
https://www.figma.com/design/8L9BXMzNTKboGWlHpdXyYP/Appsmith-IDE?node-id=33-3546&t=qpptzxhkWatrFIn1-4

Fixes [#37248](https://github.com/appsmithorg/appsmith/issues/37248)

## Automation

/ok-to-test tags="@tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/11796204902>
> Commit: 0e57343a6c2b7543f37e37940f8447be10ccd9e9
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=11796204902&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Sanity`
> Spec:
> <hr>Tue, 12 Nov 2024 12:12:33 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Updated icons for various settings-related components from "settings-2-line" to "settings-v3" to enhance visual representation.

- **Bug Fixes**
	- Improved handling of frame ancestor settings in the AdminSettings configuration, ensuring better management of stored values.

- **Documentation**
	- No changes to the documentation were made in this release. 

- **Chores**
	- Updated multiple components to reflect the new icon, maintaining overall functionality without introducing new logic.

### CODE REVIEW

File: app/client/src/pages/Editor/AppSettingsPane/AppSettings/index.tsx [hunk 1]
```diff
@@ -126,7 +126,7 @@ function AppSettings() {
   const SectionHeadersConfig: SectionHeaderProps[] = [
     {
       id: "t--general-settings-header",
-      icon: "settings-2-line",
+      icon: "settings-v3",
```
[the remark this answers is not in the document]
Comment (ankitakinger) [reply app/client/src/pages/Editor/AppSettingsPane/AppSettings/index.tsx:129]: those are the icon provider components which should provide both icons.

File: app/client/src/pages/Editor/gitSync/components/GitChangesList/StaticChange.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const allStaticChangeDefs: Record<
   [StaticChangeKind.SETTINGS]: (status: GitStatusData) => ({
     condition: status.modified.includes("application.json"),
     message: "Application settings modified",
-    iconName: "settings-2-line",
+    iconName: "settings-v3",
```
[the remark this answers is not in the document]
Comment (ankitakinger) [reply app/client/src/pages/Editor/gitSync/components/GitChangesList/StaticChange.tsx:52]: those are the icon provider components which should provide both icons.

File: app/client/src/pages/common/SearchBar/HomepageHeaderAction.tsx [hunk 3]
```diff
@@ -92,7 +92,7 @@ const HomepageHeaderAction = ({
               });
             }}
             size="md"
-            startIcon="settings-control"
+            startIcon="settings-v3"
```
[the remark this answers is not in the document]
Comment (ankitakinger) [reply app/client/src/pages/common/SearchBar/HomepageHeaderAction.tsx:95]: those are the icon provider components which should provide both icons.

**Review by albinAppsmith — APPROVED** [review]

**Pull request merged**